jlm/rvsdg/FunctionType.hpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017 Nico Reißmann <nico.reissmann@gmail.com>
+ * Copyright 2025 Helge Bahmann <hcb@chaoticmind.net>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_RVSDG_FUNCTION_TYPE_HPP
+#define JLM_RVSDG_FUNCTION_TYPE_HPP
+
+#include <jlm/rvsdg/type.hpp>
+#include <jlm/util/common.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace jlm::rvsdg
+{
+
+/**
+ * \brief Function type class
+ *
+ * Represents the type of a callable function.
+ */
+class FunctionType final : public ValueType
+{
+public:
+  ~FunctionType() noexcept override;
+
+  FunctionType(
+      std::vector<std::shared_ptr<const jlm::rvsdg::Type>> argumentTypes,
+      std::vector<std::shared_ptr<const jlm::rvsdg::Type>> resultTypes);
+
+  inline const std::vector<std::shared_ptr<const jlm::rvsdg::Type>> &
```

Comment:
minor nitpick: remove inline

Reply:
nodiscard

Reply:
why not writing out inline? I tend to use it to mark things that are really explicitly marked to be inlined (it is not required, similar to override, but documents the intent)